### DISCUSSION TITLE
How to calculate the probability value of a word?

### DISCUSSION THREAD

**Discussion by tiennguyen12g** [openai/whisper#1587] [General]
Hi everyone, I am new here, so I do not know how authors can calculate the value of probability. I guess they use one of the following methods:

1. Comparing the audio of a word in audio with a lot of sample sounds. Then, they count how many times it's correct and calculate the percentage.
   Example: With 100 sample sounds, if the word in the audio matches 50 sample sounds, the probability value is 50%.
   
2. Comparing the syllables of the word.
   Example: For the word 'contribute' - /kənˈtrɪb.juːt/. They check if the phoneme sounds are correct. If /kən/ sounds like /kan/ or /kon/, the word will only get 66.6% probability.

** If they are using the second approach, could you please explain how I can identify the incorrect phoneme?

I would like to receive results like:
- "birch": /bir/ may lack confidence, so its color is yellow, and /ch/ is the correct phoneme, so its color is green.
- "planks": /pl/ may lack confidence, so its color is green, and /anks/ is the correct phoneme, so its color is red.
I came across this picture in a discussion, but it doesn't have a solution.
![image](https://github.com/openai/whisper/assets/40747229/445b86ee-7dd0-460e-817e-5f7f2e2d6ffc)
Examples: "Agricultural," "depletion."
![whisperccp1](https://github.com/openai/whisper/assets/40747229/2b46fc04-9ec5-4b5f-a936-22abd828f36d)
I would greatly appreciate your assistance. Thank you so much.

I see the website name "apeuni" has this feature, you can see more example:

![ape2](https://github.com/Softcatala/whisper-ctranslate2/assets/108709931/22c43fd4-2e94-4c16-bdbd-0040aed9d225)
![ape1](https://github.com/Softcatala/whisper-ctranslate2/assets/108709931/e26db59f-06a0-4c86-b010-46afd6c91302)

**Comment by phineas-pta** [comment]
from what i read, i assume u have no background in machine learning

for the word probability, it requires some solid math knowledge and familiar with the transformer architecture of neural networks

giáo trình tiếng việt nếu bạn cần: https://d2l.aivivn.com/chapter_attention-mechanisms/index_vn.html

**Reply by tiennguyen12g** [reply]
Thank bro, I am coding Nodejs. So, I am just learn a little bit about Python.
